data-products/src/main/scala/org/sunbird/analytics/util/TextBookUtils.scala

```diff
@@ -32,25 +45,50 @@ object TextBookUtils {
-    val reportTuple = for {textbook <- sc.parallelize(textbookInfo)
+  def getTextbookHierarchy(config: Map[String, AnyRef], textbookInfo: List[TextBookInfo],tenantInfo: RDD[TenantInfo],restUtil: HTTPClient)(implicit sc: SparkContext, fc: FrameworkContext): (RDD[FinalOutput]) = {
+    val reportTuple = for {textbook <- textbookInfo
       baseUrl = s"${AppConf.getConfig("hierarchy.search.api.url")}${AppConf.getConfig("hierarchy.search.api.path")}${textbook.identifier}"
       finalUrl = if("Live".equals(textbook.status)) baseUrl else s"$baseUrl?mode=edit"
       response = RestUtil.get[ContentDetails](finalUrl)
       tupleData = if(null != response && "successful".equals(response.params.status)) {
       val data = response.result.content
+        val dceDialcode = generateDCEDialCodeReport(data)
+        val dceDialcodeReport = dceDialcode._1
+        val etbDialcode = generateETBDialcodeReport(data)
+        val etbDialcodeReport = etbDialcode._1
+        val dialcodeScans = dceDialcode._2 ++ etbDialcode._2
         val etbReport = generateETBTextbookReport(data)
         val dceReport = generateDCETextbookReport(data)
-        (etbReport, dceReport)
+        (etbReport, dceReport, dceDialcodeReport, etbDialcodeReport, dialcodeScans)
        }
-       else (List(),List())
+       else (List(),List(),List(),List(),List())
     } yield tupleData
     val etbTextBookReport = reportTuple.filter(f => f._1.nonEmpty).map(f => f._1.head)
     val dceTextBookReport = reportTuple.filter(f => f._2.nonEmpty).map(f => f._2.head)
-    generateTextBookReport(etbTextBookReport, dceTextBookReport, tenantInfo)
+    val dceDialCodeReport = reportTuple.map(f => f._3).filter(f => f.nonEmpty)
+    val dcereport = if(!dceDialCodeReport.isEmpty) dceDialCodeReport.head else List()
+    val etbDialCodeReport = reportTuple.map(f => f._4).filter(f => f.nonEmpty)
+    val etbreport = if(!etbDialCodeReport.isEmpty) etbDialCodeReport.head else List()
+    val dialcodeScans = reportTuple.map(f => f._5).head
+
+    generateWeeklyScanReport(config, dialcodeScans)
+    generateTextBookReport(sc.parallelize(etbTextBookReport), sc.parallelize(dceTextBookReport), sc.parallelize(dcereport), sc.parallelize(etbreport), tenantInfo)
   }
 
-  def generateTextBookReport(etbTextBookReport: RDD[ETBTextbookData], dceTextBookReport: RDD[DCETextbookData], tenantInfo: RDD[TenantInfo]): RDD[FinalOutput] = {
+  def generateWeeklyScanReport(config: Map[String, AnyRef], dialcodeScans: List[WeeklyDialCodeScans])(implicit sc: SparkContext, fc: FrameworkContext) {
+    implicit val sqlContext = new SQLContext(sc)
+    import sqlContext.implicits._
+
+    val configMap = config("reportConfig").asInstanceOf[Map[String, AnyRef]]
+    val reportConfig = JSONUtils.deserialize[ReportConfig](JSONUtils.serialize(configMap))
+    val scansDf = sc.parallelize(dialcodeScans).toDF()
+
+    reportConfig.output.map { f =>
+      CourseUtils.postDataToBlob(scansDf,f,config)
+    }
+  }
```

Comment:
Why should this data be saved to blob store?

Reply:
Since the merge script works only for merging two files.